content/blog/getting-started-with-hpe-greenlake-api-for-backup-and-recovery.md

```diff
@@ -0,0 +1,591 @@
+
+4. The API execution was executed successfully. I validated this completion using the task id that I obtained from the response headers as shown above. The API used for this: `GET /data-services/v1beta1/async-operations/{{taskId}}`
+
+> **Note**: that that the execution of this API will trigger a protection execution right after the execution of this API completed.
+
+![Task list to display completion of the application of protection policy against a VM producing protection jobs](/img/api-task-list-for-protection-policy-application.png)
+
+## Triggering a protection
+
+Once the protection policy named "VMware create three tiers" was bound to the virtual machine as shown above, I then issued a trigger to create cloud protection against the virtual machine `“0-Linux-Demo-VM02”` to create one-off cloud-protection.
+
+The below list detailed the required steps:
+
+1. I figured out the protection-job-id that is associated with `“0-Linux-Demo-VM02”`. and the  cloud backup schedule Id of the virtual machine. To achieve that, I used the HPE GreenLake [API](`https://developer.greenlake.hpe.com/docs/greenlake/services/backup-recovery/public/openapi/backup-recovery-public-v1beta1/operation/DataManagementJobsList`) for Backup and Recovery `GET /backup-recovery/v1beta1/protection-jobs` and `filter assetInfo/id eq "{VM-id}”` as shown below. Note that the variable `{vmId}` contained the value of the virtual machine id as discovered in previous step, namely `"<virtual-machine-id>"`. The response body’s JSON structure contained the id of the protection job associated with `“0-Linux-Demo-VM02”`. From the same response body, I recognized that cloud protection is the `scheduleId no 3`. The API used for this:
+
+   ```shellsession
+   GET /backup-recovery/v1beta1/protection-jobs?filter=assetInfo/id eq {{vmId}}&select=assetInfo,id,operational,protections
+   ```
+
+![API to figure out protection-jobs](/img/api-to-figure-out-protection-jobs.png)
+
+2. This use case is equivalent to run now button at the schedule of the `“0-Linux-Demo-V02”` on the selected cloud protection as shown in below figure. After clicking `“Run Now”` button, a cloud protection will commence against the virtual machine, a recovery point is going to be created at the cloud protection store. On next step, I will invoke `"Run Now"` on the `HPE Cloud Protection Store` using HPE GreenLake API for Backup and Recovery.
+
+![UI for run now protection jobs](/img/gui-run-now-cloud-protection.png)
+
+3. The GreenLake API to accomplish the use case above was `POST /backup-recovery/v1beta1/protection-jobs/:id/run`, and the documentation of this [API](https://developer.greenlake.hpe.com/docs/greenlake/services/backup-recovery/public/openapi/backup-recovery-public-v1beta1/operation/DataManagementJobRun) also list the required JSON structure for the request body. I created a JSON request body structure as the following:
```

Comment:
Add "HPE" in front of "GreenLake API". Add the word "used" prior to "to accomplish".
Change "list" to "lists".

Change "as the following" to "as follows:"